File: exercise_1/template/Solution1.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class Solution1 {

    static String solveTask1(char[][] inputMatrix) {
        /*
        Enter your code here
        Return the minimum cost or return No path found!
         */
        return "";
    }
    /*
    Use as many helper functions as you like
     */

    public static void main(String[] args) throws FileNotFoundException {
        /* Read the input matrix*/
        String filename = args[0];
        char [][] inputMatrix = readFile(filename);

        /* Your main function to solve the problem*/
        String out = solveTask1(inputMatrix);
        System.out.println(out);
    }


    public static char[][] readFile(String filepath) throws FileNotFoundException {
        ArrayList<ArrayList<String>> array = new ArrayList<>();
        Scanner input = new Scanner(new File(filepath));
        while(input.hasNextLine())
        {
            Scanner colReader = new Scanner(input.nextLine());
            ArrayList col = new ArrayList();
            while(colReader.hasNext("[*_RX]"))
            {
                col.add(colReader.next());
            }
	    if(col.size() > 0){
            	array.add(col);
	    }
        }
        input.close();

        char[][] inputMatrix = new char[array.size()][array.get(0).size()];
        for(int i=0;i<array.size();i++) {
            for(int j=0;j<array.get(0).size();j++){
                inputMatrix[i][j] = array.get(i).get(j).toCharArray()[0];
            }
        }
        return inputMatrix;
    }
}

File: exercise_1/template/Solution1.py
import sys
import numpy as np


def solve_task1(input_matrix):
	# Enter your code here.
	# Return the minimum cost or return No path found!
    return ""


# Use as many helper functions as you like


# Get input from command prompt and run the program
input_arg = sys.argv[1]
def run_program(file_name = input_arg):
    # Read the input matrix
    input_matrix = np.genfromtxt(file_name, dtype='str')
    
    # Your main function to solve the matrix
    print(solve_task1(input_matrix))


run_program()

# To test the result yourself,
# Open the command line tool, navigate to the folder and execute:
# python Solution1.py input_for_task1.txt


